python/cudf_polars/cudf_polars/experimental/io.py

```diff
@@ -47,3 +53,271 @@ def _(
         }
 
     return ir, {ir: PartitionInfo(count=1)}
+
+
+class ScanPartitionFlavor(IntEnum):
+    """Flavor of Scan partitioning."""
+
+    SINGLE_FILE = enum.auto()  # 1:1 mapping between files and partitions
+    SPLIT_FILES = enum.auto()  # Split each file into >1 partition
+    FUSED_FILES = enum.auto()  # Fuse multiple files into each partition
+
+
+class ScanPartitionPlan:
+    """Scan partitioning plan."""
+
+    __slots__ = ("factor", "flavor")
+    factor: int
```

Comment:
The meaning of factor depends on whether we are splitting or fusing files.

- If we are splitting files: The factor is the number of partitions per file.
- If we are fusing files: The factor is the number of files per partition.
- Otherwise: The factor should be `1`

Reply:
Added a note on this in the docstring.